FTBFS: unused_result in run_mir.cpp

Introduced in #2660 

https://github.com/MirServer/mir/blob/c308112149638d177b7979e0802becd3a19c9b8e/src/server/run_mir.cpp#L161-L163

```
/<<PKGBUILDDIR>>/src/server/run_mir.cpp: In function ‘void {anonymous}::fatal_signal_cleanup(int, siginfo_t*, void*)’:
/<<PKGBUILDDIR>>/src/server/run_mir.cpp:163:32: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Werror=unused-result]
  163 |         static_cast<void>(write(STDERR_FILENO, warning, len));
      |                           ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Urm.

```
/<<PKGBUILDDIR>>/src/server/run_mir.cpp:165:20: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Werror=unused-result]
  165 |         (void)write(STDERR_FILENO, warning, len);
      |               ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

--- a/src/server/run_mir.cpp
+++ b/src/server/run_mir.cpp
@@ -161,8 +161,10 @@ extern "C" [[noreturn]] void fatal_signal_cleanup(int sig, siginfo_t* info, void
         constexpr char const* warning = "!!! Fatal signal received. Attempting cleanup, but deadlock may occur";
         constexpr size_t len = std::char_traits<char>::length(warning);
         // We cannot possibly do anything sensible if this write fails, so ignore the return value.
-        // Note: static_cast<void>() is *not* sufficient to silence this warning on all compilers
-        (void)write(STDERR_FILENO, warning, len);
+        // Note: static_cast<void>() is *not* sufficient to silence this warning on all compilers,
+        // nor is (void) without storing it in a variable.
+        int n = write(STDERR_FILENO, warning, len);
+        (void)(n);
     }
     perform_emergency_cleanup();
 

